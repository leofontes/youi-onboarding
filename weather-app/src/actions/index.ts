import * as tmdbActions from './tmdbActions';
import * as youtubeActions from './youtubeActions';

export {
  tmdbActions as tmdb,
  youtubeActions as youtube,
};
