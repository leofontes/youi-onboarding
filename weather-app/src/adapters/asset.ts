export interface Asset {
  images: AssetImages;
  thumbs: AssetImages;
  title: string;
  details: string;
  extra: string;
  similar: Asset[];
  key: string;
  id: number | string;
  youtubeId: string;
  type: AssetType;
  genres?: { id: number; name: string }[];
  runtime: number;
  seasons: number;
  episodes: number;
  live?: Live;
  releaseDate: string;
}

interface Live {
  duration: number;
  elapsed: number;
  channel: { name: string; width: number };
  streams: {
    uri: string;
    type: string;
    id: string;
  }[];
}

export interface AssetImages {
  Poster: string;
  Backdrop: string;
}

export enum AssetType {
  TV = 'tv',
  MOVIE = 'movie',
  AD = 'ad',
}
