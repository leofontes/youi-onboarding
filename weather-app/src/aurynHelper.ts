import { DeviceInfo, Ref } from '@youi/react-native-youi';
import { NativeModules, findNodeHandle } from 'react-native';
import { once } from 'lodash';

const systemName = DeviceInfo.getSystemName();
const { Cloud, RefUtils } = NativeModules;

interface AurynHelper {
  hasHardwareBackButton: boolean;
  isRoku: boolean;
  updateCloudScene: (component: React.RefObject<any>) => void;
  togglePointerEvents: (ref: Ref, enabled: boolean) => boolean;
  toggleParentPointerEvents: (ref: Ref, enabled: boolean) => boolean;
}

export const AurynHelper: AurynHelper = {
  hasHardwareBackButton: !['iOS'].includes(systemName),
  isRoku: Cloud.isCloudServer,
  updateCloudScene: (component) => {
    if (!Cloud.isCloudServer || !component.current) return;
    setTimeout(() => {
      Cloud.exportSubtree(findNodeHandle(component.current), false);
      Cloud.sendFocusMap();
    }, 300);
  },
  togglePointerEvents: once((ref: Ref, enabled: boolean) => {
    RefUtils.setPointerEvents(findNodeHandle(ref), enabled);
    return true;
  }),
  toggleParentPointerEvents: once((ref: Ref, enabled: boolean) => {
    RefUtils.setParentCompositionPointerEvents(findNodeHandle(ref), enabled);
    return true;
  }),
};
