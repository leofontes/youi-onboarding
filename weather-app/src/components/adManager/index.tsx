import React, { Component, createRef, RefObject } from 'react';
import { View, StyleSheet } from 'react-native';

import PauseAd from './pause-ad';
import { AdContextType, AdContext, AdState } from './context';
import LowerThirdAd from './lower-third-ad';

interface AdManagerProps {
  pauseAdCompositionName?: string;
  pauseAdClosed: () => void;
}

class AdManager extends Component<AdManagerProps> {
  declare context: AdContextType;

  static contextType = AdContext;

  static defaultProps = {
    pauseAdClosed: () => {},
  };

  private pausAdRef: RefObject<PauseAd> = createRef();

  handleOnClose = () => this.context.setPauseAdState(AdState.closed);

  renderPauseAd = () => {
    const { pauseAdCompositionName } = this.props;

    if (pauseAdCompositionName === undefined) return null;

    this.context.setPauseAdEnabled(true);

    return <PauseAd ref={this.pausAdRef} name={pauseAdCompositionName} onClose={this.handleOnClose} />;
  };

  renderLowerThirdAd = () => {
    return <LowerThirdAd />;
  }

  render() {
    return <View style={styles.container}>
      {this.renderPauseAd()}
      {this.props.children}
      {this.renderLowerThirdAd()}
    </View>;
  }
}

const styles = StyleSheet.create({
  container: {
    position: 'absolute',
    ...StyleSheet.absoluteFillObject,
  },
});

export default AdManager;
