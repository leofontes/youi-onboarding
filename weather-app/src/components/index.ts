export { BackButton } from './backButton';
export { DiscoverContainer } from './discoverContainer';
export { List } from './list';
export { ListItem } from './listitem';
export { Timeline } from './timeline';
export { ToggleButton } from './toggleButton';
export { ToggleGroup } from './toggleGroup';
export { NavigationBar } from './navigationBar';
export { TvContainer } from './tvContainer';
export { Error } from './error';
export { withOrientation } from './withOrientation';
export { LiveContainer } from './liveContainer';
export { LiveListItem } from './liveListitem';
export { BackgroundImage } from './backgroundImage';
