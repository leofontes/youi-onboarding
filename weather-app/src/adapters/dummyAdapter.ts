import { Asset, AssetType } from './asset';

const fromApi = (fillSimilar = true): Asset => ({
  images: {
    Poster: 'https://via.placeholder.com/200x300',
    Backdrop: 'https://via.placeholder.com/1068x600',
  },
  thumbs: {
    Poster: 'https://via.placeholder.com/400x600',
    Backdrop: 'https://via.placeholder.com/534x300',
  },
  title: 'You.i TV',
  details:
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.',
  extra: 'Director: Alan Smithee | Starring: John Doe, Jane Doe',
  similar: fillSimilar ? Array(5).fill(fromApi(false)) : [],
  key: 'keystring',
  id: 1234567,
  youtubeId: 'nO_DIwuGBnA',
  type: AssetType.TV,
  genres: [28, 12],
});

export { fromApi };
