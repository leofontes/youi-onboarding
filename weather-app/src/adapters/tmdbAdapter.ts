/* eslint-disable camelcase */
import { Asset, AssetType } from './asset';
import { genres } from '../data/genres';

interface TmdbCredits {
  crew: TmdbCrew[];
  cast: TmdbCrew[];
}

interface TmdbCrew {
  job: string;
  name: string;
}

export interface TmdbResults<T> {
  results: T[];
}

interface TmdbVideo {
  key: string;
}

export interface TmdbApi {
  poster_path: string;
  backdrop_path: string;
  name: string;
  title: string;
  overview: string;
  credits: TmdbCredits;
  id: number;
  videos: TmdbResults<TmdbVideo>;
  similar: TmdbResults<TmdbApi>;
  original_language: string;
  adult: boolean;
  media_type: 'tv' | 'movie';
  type: 'tv' | 'movie';
  genres: { id: number; name: string }[];
  genre_ids: number[];
  runtime: number;
  number_of_seasons: number;
  number_of_episodes: number;
  release_date?: string;
  first_air_date?: string;
}

const getFeaturedText = (credits: TmdbCredits): string => {
  if (!credits) return '';

  const director = credits.crew.find((it: TmdbCrew) => it.job === 'Director');
  const screenplay = credits.crew.find((it: TmdbCrew) => it.job === 'Screenplay');
  const cast = credits.cast
    .slice(0, 2)
    .map((it: TmdbCrew) => it.name)
    .join('\nStarring\n\n');
  if (director && screenplay) return `${director.name}\nDirector\n\n${screenplay.name}\nScreenplay\n\n${cast}\nStarring`;
  else if (director) return `${director.name}\nDirector\n\n${cast}\nStarring`;
  else if (screenplay) return `${screenplay.name}\nScreenplay\n\n${cast}\nStarring`;

  return `${cast}\nStarring`;
};

const fromApi = (item: TmdbApi): Asset => {
  const type = 'name' in item ? AssetType.TV : AssetType.MOVIE;
  const itemGenres = item.genres || item.genre_ids.map(id => genres[type].find(it => it.id === id));
  return ({
    images: {
      Poster: `http://image.tmdb.org/t/p/w1280/${item.poster_path}`,
      Backdrop: `http://image.tmdb.org/t/p/w1280/${item.backdrop_path}`,
    },
    thumbs: {
      Poster: `http://image.tmdb.org/t/p/w500/${item.poster_path}`,
      Backdrop: `http://image.tmdb.org/t/p/w500/${item.backdrop_path}`,
    },
    title: item.name || item.title,
    details: item.overview,
    extra: getFeaturedText(item.credits),
    similar: item.similar && item.similar.results ? item.similar.results.map((it: TmdbApi) => fromApi(it)) : [],
    key: item.id.toString(),
    id: item.id,
    youtubeId: item.videos && item.videos.results.length ? item.videos.results[0].key : 'nO_DIwuGBnA',
    type,
    genres: itemGenres,
    runtime: item.runtime,
    seasons: item.number_of_seasons,
    episodes: item.number_of_episodes,
    releaseDate: item.release_date || item.first_air_date || ''
  });
}
export { fromApi };
