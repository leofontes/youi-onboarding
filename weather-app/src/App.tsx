import React from 'react';
import store from './store';
import Stack from './navigation';
import { FormFactor } from '@youi/react-native-youi';
import { Provider } from 'react-redux';
import { BackgroundImage } from './components';

export default class YiReactApp extends React.Component {
  render = () =>
    <Provider store={store}>
      <BackgroundImage source={!FormFactor.isHandset ? 'res://drawable/default/CES-auryn-10ft-bg.png' : 'res://drawable/default/Gradient-Bottom-2.png'}>
        <Stack />
      </BackgroundImage>
    </Provider>
}
