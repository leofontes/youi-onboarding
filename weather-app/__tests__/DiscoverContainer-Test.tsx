
import React from 'react';
import { DiscoverContainer } from '../src/components';
import { fromApi } from '../src/adapters/dummyAdapter';

import renderer from 'react-test-renderer';

const data = [fromApi(false), fromApi(false), fromApi(false)];

test('renders correctly', () => {
  const tree = renderer.create(<DiscoverContainer data={data} index={0} />).toJSON();
  expect(tree).toMatchSnapshot();
});
