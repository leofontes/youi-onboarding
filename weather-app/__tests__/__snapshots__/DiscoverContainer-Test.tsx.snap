// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "transparent",
    }
  }
>
  <View
    style={null}
  >
    <Composition
      onTimelinesLoaded={[Function]}
      source="Auryn_Container-Btn-Backdrop-Large"
    >
      <YiPushButtonView
        name="Btn-Backdrop-Large"
        onCompositionDidLoad={[Function]}
        onFocus={[Function]}
        onPress={[Function]}
        onTimelinesLoaded={[Function]}
      >
        <YiImageView
          name="Image-Dynamic"
          onTimelinesLoaded={[Function]}
          source={
            Object {
              "uri": "https://via.placeholder.com/1068x600",
            }
          }
        />
        <YiTextSceneNode
          name="Text-Title"
          style={
            Object {
              "color": "#f6f1ee",
            }
          }
          text="You.i TV"
        />
        <YiTextSceneNode
          name="Text-Metadata"
          text="tv"
        />
        <YiTextSceneNode
          name="Unfocus-Text-Title"
          text="You.i TV"
        />
        <YiTextSceneNode
          name="Text-Details"
          text="Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua."
        />
      </YiPushButtonView>
    </Composition>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "flex-start",
      }
    }
  >
    <View
      style={null}
    >
      <Composition
        onTimelinesLoaded={[Function]}
        source="Auryn_Container-Btn-Backdrop-Small"
      >
        <YiPushButtonView
          name="Btn-Backdrop-Small"
          onCompositionDidLoad={[Function]}
          onFocus={[Function]}
          onPress={[Function]}
          onTimelinesLoaded={[Function]}
        >
          <YiImageView
            name="Image-Dynamic"
            onTimelinesLoaded={[Function]}
            source={
              Object {
                "uri": "https://via.placeholder.com/534x300",
              }
            }
          />
          <YiTextSceneNode
            name="Text-Title"
            style={
              Object {
                "color": "#f6f1ee",
              }
            }
            text="You.i TV"
          />
          <YiTextSceneNode
            name="Text-Metadata"
            text="tv"
          />
          <YiTextSceneNode
            name="Unfocus-Text-Title"
            text="You.i TV"
          />
        </YiPushButtonView>
      </Composition>
    </View>
    <View
      style={null}
    >
      <Composition
        onTimelinesLoaded={[Function]}
        source="Auryn_Container-Btn-Backdrop-Small"
      >
        <YiPushButtonView
          name="Btn-Backdrop-Small"
          onCompositionDidLoad={[Function]}
          onFocus={[Function]}
          onPress={[Function]}
          onTimelinesLoaded={[Function]}
        >
          <YiImageView
            name="Image-Dynamic"
            onTimelinesLoaded={[Function]}
            source={
              Object {
                "uri": "https://via.placeholder.com/534x300",
              }
            }
          />
          <YiTextSceneNode
            name="Text-Title"
            style={
              Object {
                "color": "#f6f1ee",
              }
            }
            text="You.i TV"
          />
          <YiTextSceneNode
            name="Text-Metadata"
            text="tv"
          />
          <YiTextSceneNode
            name="Unfocus-Text-Title"
            text="You.i TV"
          />
        </YiPushButtonView>
      </Composition>
    </View>
  </View>
</View>
`;
